### PR TITLE
controller deployment

### PR DESCRIPTION
**Pull request by pohly**
Based on PR #1017 with one additional commit for running the controller inside a Deployment instead of StatefulSet.

### CODE REVIEW

**Comment by pohly** [comment]
This needs further work:
```
04:27:48.150  Jul 13 18:32:42.612: FAIL: Giving up waiting for PMEM-CSI to start up, check the previous warnings and log output. Last error: mutating webhook was not called for pod "#&^^%$%@%!@$@$##% invalid name", request count still 0
04:27:48.155  [1mSTEP[0m: checking for test "operator-direct-production driver late binding unsets unsuitable selected node with three replicas" in namespace kube-system, test failure
```

The request count must be summed up across all pods.

**Comment by pohly** [comment]
I force-pushed to trigger a branch build in Jenkins.

All of the Jenkins build for the base PRs failed last night because Jenkins was overloaded. My hope now is that we'll get a clean build for this branch. If so, we can merge it and thus also everything else.

**Review by avalluri — APPROVED** [review]
Looks good to me.

File: test/e2e/metrics/metrics.go [hunk 1]
```diff
@@ -40,7 +40,11 @@ var _ = deploy.Describe("direct-testing", "direct-testing-metrics", "", func(d *
 	BeforeEach(func() {
 		cluster, err := deploy.NewCluster(f.ClientSet, f.DynamicClient, f.ClientConfig())
 		framework.ExpectNoError(err, "get cluster information")
-		metricsURL = deploy.WaitForPMEMDriver(cluster, d)
+		metricsURLs := deploy.WaitForPMEMDriver(cluster, d, 1 /* controller replicas */)
+		if len(metricsURL) != 1 {
+			framework.Failf("expected one controller, got multiple metrics URLs: %v", metricsURLs)
```
Comment (pohly) [inline test/e2e/metrics/metrics.go:45]: Argh, typo: s/len(metricsURL)/len(metricsURLs)/.

Found when testing locally. Will fix before merging...

**Pull request merged**